cmd/restic/cmd_stats.go

```diff
@@ -135,6 +135,13 @@ func runStats(gopts GlobalOptions, args []string) error {
 				return fmt.Errorf("blob %v not found", blobHandle)
 			}
 			stats.TotalSize += uint64(pbs[0].Length)
+			if repo.Config().Version >= 2 {
+				if pbs[0].IsCompressed() {
+					stats.TotalUncompressedSize += uint64(pbs[0].UncompressedLength)
+				} else {
+					stats.TotalUncompressedSize += uint64(pbs[0].Length)
```

Comment:
This counts two slightly different things for compressed and uncompressed blobs:
For compressed blobs, the uncompressed length is the length of a blob after decrypting and uncompressing it. For uncompressed blobs it is the size of the encrypted blobs including the encryption overhead. It would probably make more sense to count the size of the decrypted blob. For that you can replace both cases with `pbs[0].DataLength()`.

Reply:
Oh, I didn't notice that. My idea was to show the space the repository would have occupied had compression not been enabled though, so the value should take into account the encryption overhead, and I'd like the numbers to stay the same if a repository is migrated to version 2 but not pruned, so the calculation shouldn't change for uncompressed blobs. What about using ```crypto.CiphertextLength(int(pbs[0].UncompressedLength))``` like in the commit I just pushed?

Reply:
You mean `crypto.CiphertextLength(int(pbs[0].Length))`? In that case replacing both branches with `crypto.CiphertextLength(int(pbs[0].DataLength()))` would be simpler.

So essentially that definition of uncompressed size allows you to compute the compression ratio, while also accounting for overheads due to encryption etc.